deploy/kubedirector/crd-settings.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: kubedirectorsettingss.kubedirector.bluedata.io
+spec:
+  group: kubedirector.bluedata.io
+  names:
+    kind: KubeDirectorSettings
+    listKind: KubeDirectorSettingsList
+    plural: kubedirectorsettingss
+    singular: kubedirectorsettings
+  scope: Namespaced
+  version: v1alpha1
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          properties:
+            storageClassName:
+              type: string
+              minLength: 1
+              pattern: '^[a-z0-9]([-a-z0-9]*[a-z0-9])?$'
```

Comment:
If this is the correct pattern for storage class name, we also need it in crd-cluster.yaml.

Reply:
this is the regex used by k8s when creating a storageClass

'[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'

```The StorageClass "--1123ssd" is invalid: metadata.name: Invalid value: "--1123ssd": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')```

Reply:
We still need the pattern in crd-cluster.yaml.

Reply:
forgot to commit that file